Zend/zend_compile.c

```diff
@@ -2974,7 +2988,234 @@ void zend_compile_static_call(znode *result, zend_ast *ast, uint32_t type) /* {{
 }
 /* }}} */
 
-void zend_compile_new(znode *result, zend_ast *ast) /* {{{ */
+zend_string* zend_name_anon_class(zend_ast *parent TSRMLS_DC) {
+    size_t len;
+    char   *val;
+    zend_string *anon;
+    uint32_t next = get_next_op_number(CG(active_op_array));
+    
+    if (parent) {
+        zval *extends = zend_ast_get_zval(parent);
+        len = zend_spprintf(
+            &val, 0, "%s@%p",
+            Z_STRVAL_P(extends), &CG(active_op_array)->opcodes[next-1]);
+        anon = zend_string_init(val, len, 1);
+        Z_DELREF_P(extends); /* ?? */
+        efree(val); 
+    } else {
+        len = zend_spprintf(
+            &val, 0, "class@%p", 
+            &CG(active_op_array)->opcodes[next-1]);
+        anon = zend_string_init(val, len, 1);
```

Comment:
Why allocate a persistent string here? I'd also suggest to directly print to a zend_string using `zend_strpprintf`.